contents/security/Https.md

```diff
@@ -0,0 +1,62 @@
+
+Thus, by using https, we can be sure that even though our network packets are transmitted over an insecure network, sniffers cannot understand the content of our encrypted packets. 
+
+Besides providing secure network traffic, HTTPS also provides server validation through Certificate Authority(CA) architecture. A detailed explanation on CA is [here](https://www.globalsign.com/en-sg/ssl-information-center/what-are-certification-authorities-trust-hierarchies/) . In short, CA works by issuing the server a digital certificate that can only be produced by CA. When the server sends its digital certificate to the client, client browsers verify the digital certificate with CA to check whether the server is indeed the intended server. To obtain such digital certificate, the server needs to apply to CA and CA will verify the server before issuing the digital certificate.
+
+## How to set up HTTPS
+In order set up HTTPS on your server, you would need to have:
+
+1. A dedicated IP address for your server.
+2. Obtain a digital certificate from a CA.
+
+For example, one of the CA is [digicert](https://www.digicert.com) and they provide SSL [installation guide](https://www.digicert.com/ssl-certificate-installation.htm) for different platforms. Usually, the SSL setup could be found on the CA that you obtained a digital signature from.
+
+However, most CAs are not free of charge. One free initiative to provide free domain validation certificates is [Lets Encrypt](https://letsencrypt.org/).
+
+## Misuse of HTTPS
+
+Up to this moment, it seems that nothing could go wrong with HTTPS. However, in real life, there could be weakness on the implementation of HTTPS.
+
+### SHA-1 Collision Attack
+In the implementation of HTTPS, the CA does not sign the digital certificate directly, rather it signs the fixed length [hashed digest](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the digital certificate for efficiency. This introduces the possibility of two different digital certificates with the same hashed digest. Thus, if the attackers manage to forge a fake digital certificate with the same hashed digest of another valid digital certificate, the browser would trust the attackers server and all the servers signed by the attackers. This loophole occurs with SHA-1 hashing algorithm and SHA-1 is no longer used in HTTPS after 2016. In 2017, Google has announced an algorithm to forge a duplicated SHA-1 hash and the report is [here](https://github.com/se-edu/learningresources/pull/18/files). A more detailed explanation of this problem is found [here](https://www.sott.net/article/275524-Why-HTTPS-and-SSL-are-not-as-secure-as-you-think).
```

Comment:
One of the links in the paragraph is https://github.com/se-edu/learningresources/pull/18/files; wrong link?